src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts

```diff
@@ -0,0 +1,199 @@
+        this.createConfigForm();
+        this.loading = false;
+        this.configForm.setValue(configs);
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  createConfigForm() {
+    const controlsConfig = {};
+    _.forEach(Object.values(this.options), (option) => {
+      controlsConfig[option.name] = [option.default_value, this.getValidators(option)];
+    });
+    this.configForm = this.formBuilder.group(controlsConfig);
+  }
+
+  createPreviewForm() {
+    const controls = {
+      report: JSON.stringify(this.report, null, 2),
+      reportId: this.reportId,
+      licenseAgrmt: [this.licenseAgrmt, Validators.requiredTrue]
+    };
+    this.previewForm = this.formBuilder.group(controls);
+  }
+
+  download(report: object, fileName: string) {
+    this.textToDownloadService.download(JSON.stringify(report, null, 2), fileName);
+  }
+
+  getValidators(option: any): ValidatorFn[] {
+    const result = [];
+    switch (option.type) {
+      case 'int':
+        result.push(CdValidators.number());
+        result.push(Validators.required);
+        if (_.isNumber(option.min)) {
+          result.push(Validators.min(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.max(option.max));
+        }
+        break;
+      case 'str':
+        if (_.isNumber(option.min)) {
+          result.push(Validators.minLength(option.min));
+        }
+        if (_.isNumber(option.max)) {
+          result.push(Validators.maxLength(option.max));
+        }
+        break;
+    }
+    return result;
+  }
+
+  disableModule(message: string = null, followUpFunc: Function = null) {
+    this.telemetryService.enable(false).subscribe(() => {
+      if (message) {
+        this.notificationService.show(NotificationType.success, this.i18n(message));
+      }
+      if (followUpFunc) {
+        followUpFunc();
+      } else {
+        this.router.navigate(['']);
+      }
+    });
+  }
+
+  getReport() {
+    this.loading = true;
+    this.telemetryService.getReport().subscribe(
+      (resp: object) => {
+        this.loading = false;
+        this.report = resp;
+        this.reportId = resp['report']['report_id'];
+        this.createPreviewForm();
+        this.step++;
+      },
+      (_error) => {
+        this.error = true;
+      }
+    );
+  }
+
+  updateConfig() {
+    const config = {};
+    _.forEach(Object.values(this.options), (option) => {
+      const control = this.configForm.get(option.name);
+      // Append the option only if the value has been modified.
+      if (control.dirty && control.valid) {
+        config[option.name] = control.value;
+      }
+    });
+    this.mgrModuleService.updateConfig('telemetry', config).subscribe(
+      () => {
+        this.disableModule(
+          'Your settings have been applied successfully. ' +
+            'Due to security reasons the Telemetry module is now disabled until you ' +
```

Comment:
I'm not really sure if I understand you correctly. Do you want me to add the word 'privacy' to the message?

Yes sure, it's possible you accepted the license before. The message will be shown if you made changes to the configuration and so the report content is different.

Reply:
I think a better term would be `legal reasons` instead of privacy reasons. If legal reasons would be mentioned here more people would understand why it was disabled, as it's a new use case and the terms of use don't provide such a always active agreement.

Another idea would be to simply put both in one page so that it won't be disabled in the first place. The multi page setup can cause a scenario where one just changes some attributes - doesn't read the notification and leaves the page again with the idea that he has update the telemetry settings.

IMO the one page idea is something to be discussed in a follow up issue.

Reply:
I spoke with @s0nea about the wording and we concluded to use "privacy/legal reasons" instead. We explicitly disable the module after changes have been made, to avoid transmitting additional data until the user explicitly reviewed the new content and re-confirmed their consent to the terms under which this data will be used and processed.